### PR TITLE
New input plugin: trakt_calendar

### PR DESCRIPTION
**Pull request by cvium**
### Motivation for changes:
Dunno

### Config usage if relevant (new plugin or updated schema):
```
  trakt_calendar:
    account: myaccount
    start_day: -7  # number of days from today
    days: 10
    strip_dates: yes
    type: episodes
```

### TODO
- [x] Test with `what` in discover plugin
~- [ ] Structural pytest?~
- [x] Make sure a "you missed an episode X days ago" task is feasible

### Example "you missed an episode X days ago" task
```YAML
tasks:
  notify_missed:
    disable: seen  # we do not want seen to interfere
    metainfo_series: yes
    notify: bla bla
    trakt_calendar:
      type: episodes
      account: asdas
      start_day: -7
      days: 6  # Grab episodes in the last 6 days (not including today)
    if:
      - "trakt_collected or trakt_watched": reject
      - "series_season == 0": reject
    trakt_lookup:
      account: dsadsa
    crossmatch:
      from:
        - trakt_list:
            account: dsadsa
            type: shows
            list: ddsadasdas
      action: accept
      fields: [trakt_series_name]
```

### CODE REVIEW

File: flexget/config_schema.py [hunk 1]
```diff
@@ -270,6 +277,13 @@ def is_valid_template(instance):
     return get_template(instance) is not None
 
 
+@format_checker.checks('date', raises=ValueError)
```
Comment (liiight) [inline flexget/config_schema.py]: Json schema had a built in `date-time` format, can't we use that?

Comment (cvium) [inline flexget/config_schema.py]: It doesn't conform to the same standard as dateutil, which means it validates a format that dateutil can't parse...

**Comment by mfonville** [comment]
Thanks for working on this; I tried to get it working but already noticed the following things:
I had to set `"start_date: 2017-01-01"` with the `"` otherwise it is not interpreted as a string

Also I got:
```
Series `The Simpsons S28E22` with identified_by value `auto` are not supported.
```
which I had to solve by adding `identified_by: ep` but this is something I'd expect to be set by the plugin itself.

Also `days` seems to look ahead in time (so the next 7 days) instead of back in time, is that on purpose?

**Comment by cvium** [comment]
> Also days seems to look ahead in time (so the next 7 days) instead of back in time, is that on purpose?

That's the API. I'm not sure if it supports negative numbers.

> which I had to solve by adding identified_by: ep but this is something I'd expect to be set by the plugin itself.

I don't think any other plugin automatically sets this...

**Comment by liiight** [comment]
>Also days seems to look ahead in time (so the next 7 days) instead of back in time, is that on purpose?

The idea is to get shows with an upcoming broadcast date. @cvium maybe the default needs to be today?

**Comment by mfonville** [comment]
> That's the API. I'm not sure if it supports negative numbers.
Ok, I understand. It is a bit counter-intuitive for users, so it will be important to document this in an accessible way.
It would make sense to then e.g. make the days to be '1' by default.

> I don't think any other plugin automatically sets this...
hmm, but it fails with `auto`, and calendar can only provides episode. imho it would make sense that the plugin at least sets the config-parts that it cannot do without and that are static.

**Comment by tubedogg** [comment]
It's too bad Trakt doesn't offer through the API the [custom calendars](https://blog.trakt.tv/custom-calendars-9c6a8a224f01) that VIP members can create. It basically takes a list and makes a custom calendar that only includes shows on that list, instead of using watchlist/watched/collected.

That said, will this work with `crossmatch` to be able to kind of simulate that? I'd basically want to pull a list from Trakt and be able to use crossmatch to only return entries for episodes that match a series on that list.

**Comment by cvium** [comment]
It should.

**Pull request merged**